api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityService.java

```diff
@@ -16,33 +23,72 @@
+        return buildEligibilityAndEntitlement(eligibilityResponse, entitlement);
     }
 
     /**
      * Determines the eligibility for the given existing claimant. No check is made on the NINO as they already exist in the
-     * database. The eligibility status is simply checked by calling the external service.
+     * database. The eligibility status is checked by calling the external service.
+     * Claimants determined to be eligible must still either be pregnant or have children under 4,
+     * otherwise they will be ineligible.
      *
      * @param claimant the claimant to check the eligibility for
-     * @return the eligibility response for the claimant
+     * @return the eligibility and entitlement for the claimant
      */
-    public EligibilityResponse determineEligibilityForExistingClaimant(Claimant claimant) {
-        return client.checkEligibility(claimant);
+    public EligibilityAndEntitlement determineEligibilityForExistingClaimant(Claimant claimant, LocalDate cycleStartDate, PaymentCycle previousCycle) {
+        EligibilityResponse eligibilityResponse = client.checkEligibility(claimant);
+        PaymentCycleVoucherEntitlement entitlement = cycleEntitlementCalculator.calculateEntitlement(
+                Optional.ofNullable(claimant.getExpectedDeliveryDate()),
+                eligibilityResponse.getDateOfBirthOfChildren(),
+                cycleStartDate,
+                previousCycle.getVoucherEntitlement());
+        return buildEligibilityAndEntitlement(eligibilityResponse, entitlement);
+    }
+
+    private EligibilityResponse checkEligibilityForNewClaimant(Claimant claimant) {
+        EligibilityResponse eligibilityResponse = client.checkEligibility(claimant);
+        EligibilityStatus eligibilityStatus = eligibilityStatusCalculator.determineEligibilityStatusForNewClaim(eligibilityResponse);
+        return eligibilityResponse.toBuilder()
+                .eligibilityStatus(eligibilityStatus)
+                .build();
+    }
+
+    private EligibilityAndEntitlement buildEligibilityAndEntitlement(EligibilityResponse eligibilityResponse, PaymentCycleVoucherEntitlement entitlement) {
+        EligibilityStatus eligibilityStatus = decideEligibilityStatus(eligibilityResponse, entitlement);
+        return EligibilityAndEntitlement.builder()
+                .eligibilityStatus(eligibilityStatus)
+                .voucherEntitlement(entitlement)
+                .dateOfBirthOfChildren(eligibilityResponse.getDateOfBirthOfChildren())
+                .dwpHouseholdIdentifier(eligibilityResponse.getDwpHouseholdIdentifier())
+                .hmrcHouseholdIdentifier(eligibilityResponse.getHmrcHouseholdIdentifier())
+                .build();
     }
+
+    private EligibilityStatus decideEligibilityStatus(EligibilityResponse response, PaymentCycleVoucherEntitlement voucherEntitlement) {
+        if (response.getEligibilityStatus() == EligibilityStatus.ELIGIBLE && voucherEntitlement.getTotalVoucherEntitlement() == 0) {
```

Comment:
Is this a safe way to do this, shouldn't we be checking based on the number of children and the presence of an expected delivery date, or are we presuming that the voucher entitlement process will have correctly done its job?

Reply:
We are indeed assuming that the entitlement calculator will have done its job correctly

Reply:
Yes, but my question is more whether we should when we have all the information we need without depending upon another component to have done its job, hence removing the (albeit indirect) linkage between the two components.